### ISSUE TITLE
Automatically start processes when you visit a URL

### ISSUE THREAD

**Issue by dlee** [code-mancers/invoker#122]
It would be nice if invoker could automatically start processes when a URL is visited, much like how pow automatically starts rails applications.

**Comment by gnufied** [comment]
So unlike `pow` invoker manages processes without urls as well (such as sidekiq etc). So do we propose this feature should only work for processes which are http/s in nature?

**Comment by iffyuva** [comment]
what if invoker itself is not running? in this case invoker won't be listening on port 80, and it can't start corresponding rails app if foo.dev is visited.

this feature makes more sense if invoker is running as a daemon.

**Comment by gnufied** [comment]
It also makes less Invoker less resource hungry I guess, because it won't be starting the processes until they are accessed. This is partly doable through, `disable_autorun` flag though.

**Comment by yevgenko** [comment]
:+1:  and would be nice to have processes automatically terminated after 15 minutes of inactivity

p.s.
another POW user here, lol

**Comment by yevgenko** [comment]
yes, thats mainly for the web processes, however stopping entire App (as per Procfile) still makes sense as well

**Comment by yevgenko** [comment]
on other hands, I believe, processes could be monitored and stopped by the external process manager (script) once auto-start by URL is ready
